dotnet/src/Agents/Core/Chat/KernelFunctionTerminationStrategy.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Signals termination based on the evaluation of a <see cref="KernelFunction"/>.
+/// </summary>
+/// <param name="function">A <see cref="KernelFunction"/> used for termination criteria</param>
+public class KernelFunctionTerminationStrategy(KernelFunction function) : TerminationStrategy
+{
+    /// <summary>
+    /// A well-known <see cref="KernelArguments"/> key associated with the agent name.
+    /// </summary>
+    public const string AgentArgumentName = "_agent_";
+
+    /// <summary>
+    /// A well-known <see cref="KernelArguments"/> key associated with the chat history.
+    /// </summary>
+    public const string HistoryArgumentName = "_history_";
```

Comment:
Should we make these two configurable?

Reply:
I'll be exploring this based on this comment: https://github.com/microsoft/semantic-kernel/pull/5895#discussion_r1579402184